dts/xtensa/intel/intel_cavs25_2core.dtsi

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2019,2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <xtensa/xtensa.dtsi>
+#include <dt-bindings/i2c/i2c.h>
+#include <mem.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "cdns,tensilica-xtensa-lx4";
+			reg = <0>;
+		};
```

Comment:
Not that it really matters, but FWIW: nothing in Zephyr actually uses these cpu definitions.  We still get core count out of kconfig, and to the extent we'd want to put it in devicetree, we'd want a cavs-specific node here that tells us how many cores exist.  Having multiple cpu devices only makes sense if they can have divergent configuration (e.g. the ULP core? maybe that would make sense).

Reply:
I'll keep core count also in the kconfig option for now in this PR